### DISCUSSION TITLE
Which setting for end of play

### DISCUSSION THREAD

**Discussion by mydroidbox** [pschatzmann/arduino-audio-tools#752] [Q&A] [answered]
Hey again Phil.
In your base-player-a2dp.ino sketch,
which setting determines whether bluetooth disconnects or stays connected
at the end of mp3 play.

**Comment by pschatzmann** [comment, marked as answer]
A2DP automatically disconnects if it does not receive any audio.
The AudioPlayer class has the method setSilenceOnInactive (bool active), which makes sure that the player sends 0 values instead of nothing...

**Reply by mydroidbox** [reply]
Your player works great. A while back though I was experimenting with a sketch I found in one of your discussions [here](https://github.com/pschatzmann/ESP32-A2DP/issues/108) using the 
microDecoder library as Mp3 player with your BluetoothA2DPSource and I was able to keep bluetooth connection alive at the end of play but I can't quite remember how I did it. I was able to rewind as well as start the track from the beginning after play had ended without bluetooth disconnecting but I've since lost that sketch. Are you saying there's nothing in your A2DP code itself that could allow this to happen.

**Reply by pschatzmann** [reply]
No, you need to continue to provide audio (which can be just silence)

**Reply by mydroidbox** [reply]
ok thanks Phil
